codegenerator/cli/templates/static/codegen/src/EventProcessing.res

```diff
@@ -120,7 +139,13 @@ let runEventContractRegister = (
       contextEnv
       ->ContextEnv.getAddedDynamicContractRegistrations
       ->Array.keep(({contractAddress, contractType}) =>
-        !checkContractIsRegistered(~chain, ~contractAddress, ~contractName=contractType)
+        !checkContractIsRegistered(~chain, ~contractAddress, ~contractName=contractType) &&
+        !checkContractIsInCurrentRegistrations(
+          ~dynamicContractRegistrations,
+          ~chain,
+          ~contractAddress,
+          ~contractType,
+        )
```

Comment:
Previously it would be possible for two events in a batch to register the same contract